internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets.go

```diff
@@ -32,33 +37,77 @@ type contextProviderK8sSecrets struct {
 	config *Config
 
 	clientMx sync.Mutex
-	client   k8sclient.Interface
+
+	k8sReaderProvider k8sReaderProvider
+
+	// readers holds a set of cached K8S reader per namespace.
+	// We restrict each cache to watch only 1 namespace. Watching all the namespaces would consume a lot of memory
+	// when we probably want to watch a very limited set of Secrets.
+	readers map[string]client.Reader
+
+	// ctx is the context used to start the informers. Informers will be stopped once the context is done.
+	ctx context.Context
+}
+
+// getReader returns a cached client associated to a given Kubernetes namespace.
+func (p *contextProviderK8sSecrets) getReader(namespace string) (client.Reader, error) {
```

Comment:
> Failed condition 34.1% 34.1% Coverage on New Code (is less than 40%)

I think Sonar is no happy because that function is not tested. I'll try to fix that.